### DISCUSSION TITLE
[ActionView] Object renderer gracefully fallsback to unnamespaced partials

### DISCUSSION THREAD

**Discussion by joelzwarrington** [rails/rails#53253] [Ideas]
#### Current behaviour

When `prefix_partial_path_with_controller_namespace` is true, if you're rendering an object which has a partial but not in the namespace of the controller, ActionView will be unable to find the correct partial.

As an example:
```
# app/models/post.rb
class Post < ApplicationRecord; end

# app/views/posts/_post.html
<article>
  <h1><%= post.title %></h1>
</article>

# app/controllers/admin/dashboard_controller.rb
class Admin::DashboardController
  def show
    @posts = Post.all
  end
end

# app/views/admin/dashboard/show.html.erb
<%= render @posts %>
```

The current implementation will fail to find the partial when prefix_partial_path_with_controller_namespace is enabled.

We can fix that by writing our own partial in the admin namespace
```
# app/views/admin/dashboard/posts/_post.html.erb
<article>
  <h1><%= post.title %></h1>
</article>
```

#### Requested behaviour

It would be nice if we could fallback to a partial either by:
- iteratively checking namespaced paths (e.g. `Some::Deep::Nested::NamespaceController`, try `app/views/some/deep/nested/posts/_post.html.erb` then `app/views/some/deep/posts/_post.html.erb` until the partial is found
- looking for the object partial as if `prefix_partial_path_with_controller_namespace` was false

**Comment by zzak** [comment]
If I understand correctly, there is a PR for this #50852
